capstone/src/app/components/WishlistForm.tsx

```diff
@@ -59,10 +59,33 @@ export default function WishlistForm() {
     }
   };
 
-  const removeItem = (index: number) => {
-    const updatedWishlist = wishlist.filter((_, i) => i !== index);
-    setWishlist(updatedWishlist);
-    localStorage.setItem('wishlist', JSON.stringify(updatedWishlist)); // Update localStorage
+  const removeItem = async (url: string) => {
+    try {
+      // Send DELETE request to the backend
+      const response = await fetch('/api/wishlist', {
+        method: 'DELETE',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ url }), // Send the unique URL for deletion
+      });
+  
+      if (response.ok) {
+        // Update local state and localStorage
+        setWishlist((prev) => {
+          const updatedWishlist = prev.filter((item) => item.url !== url); // Match URL for deletion
+          localStorage.setItem('wishlist', JSON.stringify(updatedWishlist));
+          return updatedWishlist;
+        });
+      } else {
+        const errorData = await response.json();
+        console.error('Error removing item:', errorData.error);
+        setError(errorData.error || 'Failed to remove item from wishlist');
+      }
+    } catch (err) {
```

Comment:
Just in case I have the incoming changes saved if you need to overwrite anything! It was from the styling and the feature of a small 'x' icon to replace the 'remove' button.